rpc/gnmi/gnmi-depth.md

```diff
@@ -0,0 +1,271 @@
+          leaf country {
+            type string;
+          }
+          
+          leaf city {
+            type string;
+          }
+        }
+      }
+
+      container description {
+        leaf fabric {
+          type string;
+        }
+      
+      }
+      container broken {
+        presence "This container is broken";
+        leaf reason {
+          type string;
+        }
+      }
+    }
+```
+
+It's tree representation:
+
+```
+module: app
+  +--rw basket
+     +--rw name?          string
+     +--rw contents*      string
+     +--rw fruits* [name]
+     |  +--rw name      string
+     |  +--rw colors*   string
+     |  +--rw size?     string
+     |  +--rw origin
+     |     +--rw country?   string
+     |     +--rw city?      string
+     +--rw description
+     |  +--rw fabric?   string
+     +--rw broken!
+        +--rw reason?   string
+```
+
+We populate this data schema with the following values:
+
+```
+    basket {
+        contents [
+            fruits
+            vegetables
+        ]
+        fruits apples {
+            size XL
+            colors [
+                red
+                yellow
+            ]
+            origin {
+                country NL
+                city Amsterdam
+            }
+        }
+        fruits orange {
+            size M
+        }
+        description {
+            fabric cotton
+        }
+        broken {
+            reason "too heavy"
+        }
+    }
+```
+
+## 3. Concepts
+
+The Depth extension allows clients to specify the depth of the subtree to be returned in the response. The depth is specified as the number of levels below the specified path.
+
+The extension itself has a single field that controls the depth level:
+
+```proto
+message Depth {
+  uint32 level = 1;
+}
+```
+
+### 3.1 Depth level values
+
+#### 3.1.1 Value 0
+
+Depth value of 0 means no depth limit and behaves the same as if the extension was not specified at all.
+
+#### 3.1.2 Value 1
+
+Value of 1 means only the specified path and its direct children will be returned. See Children section for more info.
+
+#### 3.1.2 Value of N+
+
+Value of N+ where N>1 means all elements of the specified path up to N level and direct children of N-th level.
+
+### 3.2 Children nodes
+
+The Depth extension operates the value of "direct children of a schema node". What we understand by direct children:
+
+1. leafs
+2. leaf-lists
+
+Only these elements are to be returned if depth extension with non-0 value is specified for a specified depth level.
+
+### 3.3 RPC support
+
+The Depth extension applies to Get and Subscribe requests only. When used with Capability and Set RPC the server should return an error.
+
+## 4 Examples
+
+Using the data model from Section 2 we will run through a set of examples using the patched version of [openconfig/gnmic](https://gnmic.openconfig.net/) client with the added Depth extension support. We can provide the patched gnmic binary for Linux x86_84 if you want to try it out.
```

Comment:
I'd suggest removing the "we can provide..." sentence for the specification here.

Reply:
apoligies, this was a leftover from the original issue description